Adding more stuff to the Home page - Trello

Mordax opened the pull request:
Link to the Trello request: https://trello.com/c/naQPHMmV
- [x]  Link to Favorites playlist
- [x]  Genre list ~~(collapsible)~~ **just linked it**
- [x]  Platform list (links to search results for that platform) - **This kinda took up a lot of vertical space. Feedback would be great**
- [ ]  A little Help link/text explaining the search system - **People should be encouraged to read the Readme, it never gets read**
- [ ]  Recently played games - **Would require too much rework to implement**
- [x]  Link to the Trello board of planned features

Visual screenshot: 

![visual](https://user-images.githubusercontent.com/9853904/57593274-b21ddb00-7508-11e9-8010-286419945814.png)

TBubba commented on src/renderer/components/pages/HomePage.tsx:
```diff
@@ -191,6 +233,20 @@ export class HomePage extends React.Component<IHomePageProps, IHomePageState> {
 
+  private onFavoriteClick = () => {
+    const { central, libraryData, onSelectPlaylist } = this.props;
+    let hof = findFavoritePlaylist(central.playlists.playlists);
```
"hof" stands for "Hall of Fame". Consider renaming it to something more fitting like "fav" or "favorites".

TBubba commented on src/renderer/components/pages/HomePage.tsx:
```diff
@@ -201,13 +257,61 @@ export class HomePage extends React.Component<IHomePageProps, IHomePageState> {
+  private getFavoriteBrowseRoute = (): string => {
+    const defaultLibrary = this.props.libraryData.libraries.find(library => !!library.default);
+    const defaultRoute = defaultLibrary ? joinLibraryRoute(defaultLibrary.route) : Paths.BROWSE;
+    let hof = findFavoritePlaylist(this.props.central.playlists.playlists);
```
"hof" stands for "Hall of Fame". Consider renaming it to something more fitting like "fav" or "favorites".

TBubba commented:
Thank you for contributing.

**Bugs / Issues:**
- [ ] Clicking a platform link does not deselect the current playlist

Mordax commented:
I added the specified bugs as a new issue as to not keep cluttering this PR. Awaiting response regarding Genre list from Nosamu. 

Platform list is now grabbed dynamically (although there's an extraneous comma at the end of the list which I'm not sure how to get rid of due to React mapping. )

Added some icons that I thought were relevant, let me know if you prefer something else. For some reason, the list refuses to nicely center-align like the rest of the QuickStartItems (and your demo screenshot). Added a tinted background to the platform links and removed the underlines. Most colours seemed a bit garish compared to the rest of the homepage so I just kept it a lighter grey. If you have any preference, let me know. 

Platforms are in alphabetical order.

Mordax commented:
I think I've covered all the stuff you mentioned.
